src/main/java/com/dev/cinema/util/HashUtil.java

```
@@ -0,0 +1,33 @@
+package com.dev.cinema.util;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+import org.apache.log4j.Logger;
+
+public class HashUtil {
+    private static Logger logger = Logger.getLogger(HashUtil.class);
+
+    public static byte[] getSalt() {
+        SecureRandom random = new SecureRandom();
+        byte[] salt = new byte[16];
+        random.nextBytes(salt);
+        return salt;
+    }
+
+    public static String hashPassword(String password, byte[] salt) {
+        StringBuilder hashedPassword = new StringBuilder();
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance("SHA-512");
+            messageDigest.update(salt);
+            byte[] hashedPass = messageDigest.digest(password.getBytes());
+            for(byte b: hashedPass) {
+                hashedPassword.append(String.format("%02x",b));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            logger.error(e);
```

Let's also throw an exception in here